File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Utilities/TokenTypeAnalyzer/Identifiers.cs
using System;
using System.Linq;
using AliasName = System.Exception;

public class Sample
{
    int field;

    public Sample()
    {
        var var = string.Empty;
        (var, var z) = ("a", 'x');
    }

    public void Go(int value)
    {
        var var = new Generic<int>();
        var alias = new AliasName();
        dynamic dynamic = var;
        value = 42;
    }

    public void ValueAsVariableForCoverage()
    {
        string value = null;

        foreach (var var in Enumerable.Range(0, 10)) { }
    }

    public int Property
    {
        get => field;
        set
        {
            field = value;
        }
    }
}

public class Generic<TItem> { }

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Utilities/TokenTypeAnalyzer/Identifiers.vb
Imports System
Imports AliasName = System.Exception

Public Class Sample

    Private Field As Integer

    Public Sub New()
    End Sub

    Public Sub Go(Value As Integer)
        Dim G As New Generic(Of Integer)
        Dim A As New AliasName
        Value = 42
    End Sub

    Public Sub ValueAsVariableForCoverage()
        Dim Value As String = Nothing
    End Sub

    Public Property Prop As Integer
        Get
            Return Field
        End Get
        Set(Value As Integer)
            Field = Value
        End Set
    End Property

End Class

Public Class Generic(Of ItemType)
End Class
